drivers/disk/flashdisk.c

```diff
@@ -184,35 +228,42 @@ static int update_flash_block(off_t start_addr, uint32_t size, const void *buff)
 static int disk_flash_access_write(struct disk_info *disk, const uint8_t *buff,
 				 uint32_t start_sector, uint32_t sector_count)
 {
+	struct flashdisk_data *ctx;
 	off_t fl_addr;
 	uint32_t remaining;
 	uint32_t size;
 
-	fl_addr = lba_to_address(start_sector);
+	ctx = CONTAINER_OF(disk, struct flashdisk_data, info);
+
+	fl_addr = ctx->offset + start_sector * SECTOR_SIZE;
+	if (fl_addr >= (ctx->offset + ctx->size)) {
```

Comment:
Maybe move this check to common function so the verification logic is shared with `disk_flash_access_read()`?